File: app/src/main/java/com/marcelorcorrea/falae/storage/SharedPreferencesUtils.java
package com.marcelorcorrea.falae.storage;

import android.content.Context;
import android.content.SharedPreferences;

import com.marcelorcorrea.falae.R;

/**
 * Created by bellini on 26/05/2017.
 */

public class SharedPreferencesUtils {

    private static SharedPreferences sharedPreferences;
    private static SharedPreferences.Editor editor;

    public static void storeBooleanPreferences(String key, Boolean value, Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public static void storeStringPreferences(String key, String value, Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public static void storeIntPreferences(String key, int value, Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        editor.putInt(key, value);
        editor.apply();
    }

    public static boolean getBooleanPreferences(String key, Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        return sharedPreferences.getBoolean(key, false);
    }

    public static String getStringPreferences(String key, Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        return sharedPreferences.getString(key, "");
    }

    public static int getIntPreferences(String key, Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        return sharedPreferences.getInt(key, 0);
    }
}

File: app/src/main/java/com/marcelorcorrea/falae/storage/SharedPreferencesUtils.kt
package com.marcelorcorrea.falae.storage

import android.content.Context
import android.content.SharedPreferences

import com.marcelorcorrea.falae.R

/**
 * Created by bellini on 26/05/2017.
 */

object SharedPreferencesUtils {

    private var sharedPreferences: SharedPreferences? = null

    fun storeBoolean(key: String, value: Boolean, context: Context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        sharedPreferences?.edit()?.putBoolean(key, value)?.apply()
    }

    fun storeString(key: String, value: String, context: Context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        sharedPreferences?.edit()?.putString(key, value)?.apply()
    }

    fun storeInt(key: String, value: Int, context: Context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        sharedPreferences?.edit()?.putInt(key, value)?.apply()
    }

    fun getBoolean(key: String, context: Context): Boolean {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        return sharedPreferences?.getBoolean(key, false)!!
    }

    fun getString(key: String, context: Context): String {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        return sharedPreferences?.getString(key, "")!!
    }

    fun getInt(key: String, context: Context): Int {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        return sharedPreferences?.getInt(key, 0)!!
    }

    fun remove(key: String, context: Context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
        sharedPreferences?.edit()?.remove(key)?.apply()
    }
}
